### PR TITLE
fix ingress-to-route-controller RBAC

### PR DESCRIPTION
**Pull request by atiratree**
found some missing RBAC requirements for  ingress-to-route-controller when implementing https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/255. Till now they were fulfilled by using shared kube informers

### CODE REVIEW

File: bindata/v3.11.0/openshift-controller-manager/ingress-to-route-controller-clusterrole.yaml [hunk 1]
```diff
@@ -15,7 +15,8 @@ rules:
 - apiGroups:
   - networking.k8s.io
   resources:
-  - ingress
+  - ingresses
+  - ingressclasses
```
Comment (deads2k) [inline bindata/v3.11.0/openshift-controller-manager/ingress-to-route-controller-clusterrole.yaml:19]: This looks off to me.  Can you create another cluusterrole that is used to establish watches and grant write event permissions?

The permissions for this particular controller should only be those permissions needed for the control loop, which are usually the write permissions.

File: test/framework/clientset.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func MustNewClientset(t *testing.T, kubeconfig *restclient.Config) *Clientset {
 // getConfig creates a *rest.Config for talking to a Kubernetes apiserver.
 // Otherwise will assume running in cluster and use the cluster provided kubeconfig.
 //
-// Config precedence
+// # Config precedence
```
Comment (deads2k) [inline test/framework/clientset.go:70]: why this change?

Comment (atiratree) [reply test/framework/clientset.go]: I get these changes when I run `make update` but CI doesn't agree with me, reverted...

File: bindata/v3.11.0/openshift-controller-manager/ingress-to-route-controller-clusterrole.yaml (hunk 1 above)
Comment (atiratree) [reply bindata/v3.11.0/openshift-controller-manager/ingress-to-route-controller-clusterrole.yaml:19]: we are already giving these persmissions via shared informers and `openshift-controller-manager-sa` https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/bindata/v3.11.0/openshift-controller-manager/informer-clusterrole.yaml

I am doing something similar in the split https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/d1ad600a0c86a3b808e529bb622a7b2a3e33b17e/bindata/v3.11.0/openshift-controller-manager/route-controller-informer-clusterrole.yaml

this change here was only for declaring the RBAC dependency of the `ingress-to-route-controller`
